app/src/main/java/com/example/wordsapp/LetterAdapter.kt

```diff
@@ -40,9 +40,7 @@ class LetterAdapter :
         val button = view.findViewById<Button>(R.id.button_item)
     }
 
-    override fun getItemCount(): Int {
-        return list.size
-    }
+    override fun getItemCount(): Int = list.size
```

Comment:
I think it's possible to improve even more, what do you think about that?
```suggestion
    override fun getItemCount() = list.size
```

Reply:
Pierre I think the way it as originally was better for initial understanding. Then as you rightly pointed out the condensed refactored version is how you would solve on the job. The original in my opinion helps with basic beginner understanding ie. It highlights getItemCount is a function with return type Int and the return FROM that function is the list size. With out knowing this, its possible to interpret the condensed version as a variable assignment. Then what getItemCount does with it? and why not as a parameter to the variable etc. So a whole rabbit hole of questions which pop up during the learning journey.